stl/inc/ostream

```diff
@@ -1041,7 +1041,10 @@ basic_ostream<_Elem, _Traits>& flush_emit(basic_ostream<_Elem, _Traits>& _Ostr)
     _Ostr.flush();
     const auto _Sync_buf_ptr = dynamic_cast<_Basic_syncbuf_impl<_Elem, _Traits>*>(_Ostr.rdbuf());
     if (_Sync_buf_ptr) {
-        _Sync_buf_ptr->_Do_emit();
+        const bool _Emit_failed = !_Sync_buf_ptr->_Do_emit();
+        if (_Emit_failed) {
+            _Ostr.setstate(ios_base::badbit);
+        }

```

Comment:
This is omitting the construction of a `sentry` object, but I suspect this is observable:
https://github.com/microsoft/STL/blob/2f9f567ffe1087353ecacc5a03f81c395a1922e1/stl/inc/ostream#L91-L105
(`_Sentry_base` calls `_Rdbuf->_Lock()` but I believe that's unobservable, because only `basic_filebuf` implements that, whereas here we're dealing with a known `basic_syncbuf`.)

If the `sentry` is unobservable, I'd like to understand why.